packages/core/core/src/types.ts

```diff
@@ -149,6 +149,25 @@ export interface UseUserBilling<USER_BILLING, USER_BILLING_ITEM> {
   error: ComputedProperty<UseUserBillingErrors>;
 }
 
+export interface UseNewsletterErrors {
+  updateNewsletterData?: Error;
+  load?: Error;
+}
+
+export interface NewsletterSections {
+  woman: boolean,
+  man: boolean,
+  kids: boolean,
+}
+
+export interface UseNewsletter {
+  newsletter: ComputedProperty<NewsletterSections>;
+  loading: ComputedProperty<boolean>;
+  error: ComputedProperty<UseNewsletterErrors>;
+  updateNewsletterData: (params: NewsletterSections) => Promise<NewsletterSections>;
+  load: () => Promise<NewsletterSections>;
+}
```

Comment:
I see few issues here:
* Response from the API saved in the `newsletter` property will be different for every integration, so we cannot assume any signature. We need TypeScript's generic here, like `NEWSLETTER`.
* We don't return any values from the `load` or `update` methods in any other composable, so to be consistent, both should return `Promise<void>`.
* We don't know what properties might be needed in `update` method. I would suggest using generic here, like `NEWSLETTER_UPDATE_PARAMS`.

```javascript
export interface UseNewsletter<NEWSLETTER, NEWSLETTER_UPDATE_PARAMS> {
  newsletter: ComputedProperty<NEWSLETTER>;
  loading: ComputedProperty<boolean>;
  error: ComputedProperty<UseNewsletterErrors>;
  updateNewsletterData: (params: NEWSLETTER_UPDATE_PARAMS) => Promise<void>;
  load: () => Promise<void>;
}
```